cypress/utils/commands.js

```diff
@@ -30,10 +30,49 @@ Cypress.Commands.add('getElementsByTestIds', (testIds, options = {}) => {
   return cy.get(selectors.join(','), options);
 });
 
-Cypress.Commands.add('getElementByDataTestId', (testId) => {
-  return cy.get(`[data-testid="${testId}"]`);
+/**
+ * Get DOM elements with a data-test-subj id containing the testId.
+ * @param {string} testId data-test-subj value.
+ * @param {object} options get options. Default: {}
+ * @example
+ * // returns all DOM elements that has a data-test-subj including the string 'table'
+ * cy.getElementsByTestIdLike('table')
+ */
+Cypress.Commands.add('getElementsByTestIdLike', (partialTestId, options = {}) => {
+  return cy.get(`[data-test-subj*="${partialTestId}"]`, options);
 });
 
+/**
+ * Find element from previous chained element with a data-test-subj id containing the testId.
+ * @param {string} subject DOM object to find within.
+ * @param {string} testId data-test-subj value.
+ * @param {object} options get options. Default: {}
+ * @example
+ * // returns all DOM elements that has a data-test-subj including the string 'table'
+ * cy.findElementsByTestIdLike('table')
+ */
+Cypress.Commands.add(
+  'findElementByTestIdLike',
+  { prevSubject: true },
+  (subject, partialTestId, options = {}) => {
+    return cy.wrap(subject).find(`[data-test-subj*="${partialTestId}"]`, options);
```

Comment:
why is it partialTestId? would this allow for partial match? If so what's the mechanism that makes that work? The `*`? Do you have any doc reference to this by any chance? (For my curiosity)

Reply:
This is a partial match. As long as the data-test-subj contains that partialTestId it'll find it. 

This is not a Cypress specific selector, rather a CSS selector: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

Reply:
ah! my css newb is showing! TIL thanks!